plugins/default/validNumberRange.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  HookEventType,
+  PluginContext,
+  PluginHandler,
+  PluginParameters,
+} from '../types';
+import { getText } from '../utils';
+
+function validNumberRange(
+  target: Number,
+  lowerLimit: Number,
+  upperLimit: Number
+): boolean {
+  try {
+    return target >= lowerLimit && target <= upperLimit;
+  } catch (error) {
+    return false;
+  }
+}
```

Comment:
I don't think this logic is what the expectation is. Ideally plugins are used as part of generation so we must be able to use this rule with generated content. This rule must check weather the generated content (number) is within the range.

use `getText` function to retrieve the message content and make sure the content is number (or possibly use regex to extract number from the message content) and do validation with the extracted number.

Check existing plugins here: https://github.com/Portkey-AI/gateway/tree/main/plugins/default